server/basic-server.js

```diff
@@ -1,44 +1,74 @@
 const cors = require('cors');
-
-// Middleware
 const parser = require('body-parser');
+const redis = require('redis');
+const bluebird = require('bluebird');
+const cluster = require('cluster');
+const cpuCount = require('os').cpus().length;
+const { Pool } = require('pg');
 
-const app = express();
-module.exports.app = app;
+const pool = new Pool({
+  user: 'samliebow',
+  password: 'sdc',
+  host: '127.0.0.1',
+  database: 'quickstarter',
+});
 
-// Set what we are listening on.
-app.set('port', (process.env.PORT || 3006));
+bluebird.promisifyAll(redis.RedisClient.prototype);
+bluebird.promisifyAll(redis.Multi.prototype);
 
-// Logging and parsing
-app.use(parser.json());
+const client = redis.createClient();
+client.on('error', err => console.error(err));
+client.on('connect', () => console.log(`Redis connected for ${process.pid}`));
 
-// Serve the client files
-app.use(express.static(`${__dirname}/../client/dist`));
+if (cluster.isMaster && cpuCount > 1) {
+  console.log(`Master ${process.pid} started`);
+  for (let i = 0; i < cpuCount; i++) {
+    cluster.fork();
+  }
 
-// enabling CORS requests
-// app.use((req, res, next) => {
-//   res.header('Access-Control-Allow-Origin', '*');
-//   res.header('Access-Control-Allow-Headers', 'Origin, X-Requested-With, Content-Type, Accept');
-//   next();
-// });
-app.use(cors());
+  cluster.on('exit', (worker, code) => {
+    if (code) {
+      console.log(`Worker ${worker.process.pid} killed by error, code ${code}`);
+    } else {
+      console.log(`Worker ${worker.process.pid} exited`);
+    }
+  });
+} else {
+  const app = express();
+  app.set('port', (process.env.PORT || 3006));
+  app.use(parser.json());
+  app.use(cors());
+  app.use(express.static(`${__dirname}/../client/dist`));
 
-// Handle Get requests
-app.get('/api/community/:id', (req, res) => {
-  console.log('Handling get request');
-  utils.loadProject(req.params.id, (project) => {
-    console.log(project);
-    var project = project[0];
-    console.log('Retrieving backers from project:', project);
-    utils.loadBackers(project.backerIds, (backers) => {
-      res.json([project, backers]);
-    });
+  app.get('/api/community/:id', async (req, res) => {
+    const { id } = req.params;
+    try {
+      const cache = await client.getAsync(id);
+      let result;
+      if (cache) {
+        result = JSON.parse(cache);
+      } else {
+        result = await pool.query(`
+          SELECT title, creator, backers, 
+            name, city, country, avatar, 
+            projects_backed AS "fundedProjects"
+          FROM quickstarter.projects 
+          INNER JOIN quickstarter.projects_backers 
+            ON quickstarter.projects.id = quickstarter.projects_backers.project
+          INNER JOIN quickstarter.users
+            ON quickstarter.projects_backers.backer = quickstarter.users.id
+          WHERE quickstarter.projects.id = $1::integer;`, [id]);
+        client.set(id, JSON.stringify(result), 'EX', 60);
```

Comment:
no need to set an expiration time, redis defaults to a LRU ejection method

Reply:
Thanks, I'll fix and look at fine-tuning the memory limit.